### DISCUSSION TITLE
Syntax highlighting for html and mysql inside a php file

### DISCUSSION THREAD

**Discussion by razvbir** [helix-editor/helix#4026] [Q&A]
Hi,

I'm trying to use helix for php development. I was wondering if there is anyway to get syntax highlighting for html and mysql inside a php file. I've seen a reddit thread about how to do it in [vim](https://www.reddit.com/r/vim/comments/6ld9a1/some_nice_syntax_secrets_for_php_that_i_found/) and it seems to work out of the box on sublime text. Can it be done in helix?

**Comment by kirawi** [comment]
You need to extend the tree-sitter injections for PHP: https://github.com/helix-editor/helix/blob/master/runtime/queries/php/injections.scm

**Reply by razvbir** [reply]
This is my ~/.cargo/bin/runtime/queries/php/injections.scm. I don't really know Scheme. I've tried them both with no luck. 

Variant 1:
```
((text) @injection.content
 (#set! injection.language "html")
 (#set! injection.language "sql")
 (#set! injection.combined))

((comment) @injection.content
 (#set! injection.language "comment"))
```

Variant 2:
```
((text) @injection.content
 (#set! injection.language "html")
 (#set! injection.combined))

((text) @injection.content
 (#set! injection.language "sql")
 (#set! injection.combined))

((comment) @injection.content
 (#set! injection.language "comment"))

```

**Reply by kirawi** [reply]
You can't have two captures for a `(text)` node work; only the first capture will apply. I'm not familiar with PHP so I can't write the queries for you, but you can look at these resources:

- [Query syntax](https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax)
- [Injections](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection)
- [tree-sitter playground](https://tree-sitter.github.io/tree-sitter/playground) to see the AST for a given piece of code.

**Reply by razvbir** [reply]
This was very helpful, thank you very much. I've kind of manage to get it working. 

![Screenshot_20220929_231110](https://user-images.githubusercontent.com/6582802/193132595-234d94a7-55b1-4d9d-be8a-c5d7708dd021.png)

I'm looking for ways to improve it and I'd appreciate any hints.

![Screenshot_20220929_231158](https://user-images.githubusercontent.com/6582802/193132772-f9f8323b-e40d-4bb7-a2bc-26955c9cb228.png)

**Reply by kirawi** [reply]
Feel free to open a PR.
